mono/utils/mono-threads-coop.c

```diff
@@ -648,10 +648,16 @@ mono_threads_suspend_policy (void)
 		// otherwise if there's a compiled-in default, use it.
 		// otherwise if one of the old environment variables is set, use that.
 		// otherwise use full preemptive suspend.
+
+		W32_DEFINE_LAST_ERROR_RESTORE_POINT;
+
 		   (policy = threads_suspend_policy_getenv ())
 		|| (policy = threads_suspend_policy_default ())
 		|| (policy = threads_suspend_policy_getenv_compat ())
 		|| (policy = MONO_THREADS_SUSPEND_FULL_PREEMPTIVE);
+
+		W32_UPDATE_LAST_ERROR_RESTORE_POINT;
```

Comment:
This should be `W32_RESTORE_LAST_ERROR_FROM_RESTORE_POINT`, I think?

Reply:
Oops probably. I thought there’d only be two symbols, found these two, glossed over the definition, thought the name sounded funny, couldn’t imagine a third symbol. Will look again.

Reply:
Fixed. I also suggest removing `W32_UPDATE_LAST_ERROR_RESTORE_POINT`.
It isn't used. I have preserved lasterror a fair amount through the years, using a local with constructor/destructor, and don't recall a use-case for this extra piece.